pkg/pluginmanager/manager.go

```diff
@@ -1230,6 +1231,7 @@ func doDeletePluginFromCatalog(pluginName string, target configtypes.Target, cat
 		if err != nil {
 			return fmt.Errorf("plugin %q could not be deleted from cache", pluginName)
 		}
+		c.Unlock()
```

Comment:
I think we should call the `c.Unlock()` right after the delete, even if there is an error.  The catalog file is locked even if the delete fails.